sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -316,7 +316,7 @@ class HiveContext private[hive](
   }
 
   protected[sql] override def parseSql(sql: String): LogicalPlan = {
-    super.parseSql(substitutor.substitute(hiveconf, sql))
+    sqlParser.parsePlan(substitutor.substitute(hiveconf, sql))
```

Comment:
How about gradually moving functionality from the DLL parser to SparkQl? That would allow us to test this in the meantime.